matrix-e2e.c

```diff
@@ -47,6 +52,14 @@ typedef struct _MatrixE2ERoomData {
     GHashTable *megolm_sessions_inbound;
 } MatrixE2ERoomData;
 
+typedef struct _MatrixOlmSession {
```

Comment:
Can I have multiple sessions for the same sender key - i.e. is it a hash to one entry or a hash to many? I ask because we're supposed to call olm_matches_inbound_session which makes me assume there can be multiple